pinot-tools/src/main/java/com/linkedin/pinot/tools/segment/converter/PinotSegmentConvertCommand.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.tools.segment.converter;
+
+import com.linkedin.pinot.core.data.readers.FileFormat;
+import com.linkedin.pinot.tools.AbstractBaseCommand;
+import com.linkedin.pinot.tools.Command;
+import java.io.File;
+import org.apache.commons.io.FileUtils;
+import org.kohsuke.args4j.Option;
+
+
+/**
+ * The <code>PinotSegmentConvertCommand</code> class provides tools to convert Pinot segment into other formats.
+ * <p>Currently support converting Pinot segment to:
+ * <ul>
+ *   <li>AVRO format</li>
+ *   <li>CSV format</li>
+ *   <li>JSON format</li>
+ * </ul>
+ */
+@SuppressWarnings("FieldCanBeLocal")
+public class PinotSegmentConvertCommand extends AbstractBaseCommand implements Command {
+
+  @Option(name = "-segmentDir", required = true, metaVar = "<String>", usage = "Path to Pinot segment directory.")
+  private String _segmentDir;
+
+  @Option(name = "-outputFile", required = true, metaVar = "<String>", usage = "Path to output file.")
+  private String _outputFile;
+
+  @Option(name = "-fileFormat", required = true, metaVar = "<String>", usage = "Format to convert to (AVRO/CSV/JSON).")
+  private String _fileFormat;
+
+  @Option(name = "-csvDelimiter", required = false, metaVar = "<char>", usage = "CSV delimiter (default ',').")
+  private char _csvDelimiter = ',';
+
+  @Option(name = "-csvListDelimiter", required = false, metaVar = "<char>",
+      usage = "CSV List delimiter for multi-value columns (default ';').")
+  private char _csvListDelimiter = ';';
+
+  @Option(name = "-csvWithHeader", required = false, metaVar = "<boolean>", usage = "Print CSV Header (default false).")
+  private boolean _csvWithHeader;
+
+  @Option(name = "-overwrite", required = false, metaVar = "<boolean>",
+      usage = "Overwrite the existing file (default false).")
+  private boolean _overwrite;
+
+  @Option(name = "-help", required = false, help = true, aliases = {"-h", "--h", "--help"},
+      usage = "Print this message.")
+  private boolean _help;
+
+  @Override
+  public boolean execute()
```

Comment:
Can this be enhanced to handle the following:
1. Segments in tar.gz format.
2. Case when segmentDir contains a list of segments, and not just one segment.